### PR TITLE
Refactor Donation Goals Controller Tests with helpers

### LINKED ISSUE #428: Refactor Donation Goal Controller Tests with helpers
# Problem

Adding helpers from api_case.ex to the tests for Donation Goal Controller.

## References

Progress on: #413

### PR DESCRIPTION
**Pull request by mackenziehicks**
# What's in this PR?

Added helpers from api_case.ex to the tests for Donation Goal Controller.

Added filter method to Donation Goal Controller

Removed empty struct from two User Role Controller Tests

## References
Fixes #428 

Progress on: #413

### CODE REVIEW

**Review by amyschools — APPROVED** [review]
👍

File: test/controllers/donation_goal_controller_test.exs [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
     test "lists all entries on index", %{conn: conn} do
-      path = conn |> path_for(:index)
-      json = conn |> get(path) |> json_response(200)
+      [donation_goal_1, donation_goal_2] = insert_pair(:donation_goal)
 
-      assert json["data"] == []
+      conn
+        |> request_index
+        |> json_response(200)
+        |> assert_ids_from_response([donation_goal_1.id, donation_goal_2.id])
```
Comment (joshsmith) [inline test/controllers/donation_goal_controller_test.exs]: Alignment is off here.

**Review by joshsmith — APPROVED** [review]

File: web/controllers/donation_goal_controller.ex [hunk 2]
```diff
@@ -2,12 +2,18 @@ defmodule CodeCorps.DonationGoalController do
...
 
+  import CodeCorps.Helpers.Query, only: [id_filter: 2]
+
   alias CodeCorps.DonationGoal
 
   plug :load_and_authorize_changeset, model: DonationGoal, only: [:create]
   plug :load_and_authorize_resource, model: DonationGoal, only: [:update, :delete]
   plug JaResource
 
+  def filter(_conn, query, "id", id_list) do
```
Comment (sbatson5) [inline web/controllers/donation_goal_controller.ex]: What are your thoughts on doing this in one line?

`def filter(_conn, query, "id", id_list), do: id_filter(query, id_list)`

**Review by sbatson5 — COMMENTED** [review]
Just 1 minor suggestion.  This looks great 👍

**Review by joshsmith — APPROVED** [review]
Approved pending change @sbatson5 requested.

**Pull request merged**